frontend/docs/components/ui/cookie-banner.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { cn } from "@/lib/utils";
+import { Button } from "./button";
+import { CookieIcon } from "@radix-ui/react-icons";
+import posthog from "posthog-js";
+
+import { loadReoScript } from "@/lib/reoWrapper";
+
+
+export function cookieConsentGiven() {
+    if (!localStorage.getItem('cookie_consent')) {
+        return 'undecided';
+    }
+    return localStorage.getItem('cookie_consent');
+}
+
+
+export default function CookieConsent({ variant = "default", demo = false, onAcceptCallback = () => { }, onDeclineCallback = () => { } }) {
+    const [isOpen, setIsOpen] = useState(false);
+    const [hide, setHide] = useState(false);
+    const [consentGiven, setConsentGiven] = useState('');
+
+    const accept = () => {
```

Comment:
I think this should probably be wrapped in a `useCallback` (same with the `decline` below) to avoid rerenders if we can